gufe/components/explicitmoleculecomponent.py

```diff
@@ -68,6 +72,12 @@ def _check_partial_charges(mol: RDKitMol) -> None:
                   f"RDKit formal charge {Chem.GetFormalCharge(mol)}")
         raise ValueError(errmsg)
 
+    # set the charges on the atoms if not already set
+    for i, charge in enumerate(p_chgs):
+        atom = mol.GetAtomWithIdx(i)
+        if not atom.HasProp("PartialCharge"):
```

Comment:
Similar to the next point if the molecule level charges are set I think its safe to assume they should be used and setting them to be the same should be safe, so we should not need a warning.

Reply:
I think here I'm going to ask that we do have a check for equality & raise an error if it doesn't work out - if the atomic properties don't match the molecule ones (unless RDKit has its own internal check for this), then we're in a case of undefined behaviour and everything is likely to break later on when we do the `to_dict` and `from_dict` calls.

Reply:
May the "explain why I'm thinking this": there's a lot of rather interesting ways for charge setting and a recent deep dive seems to indicate that none of it is consistent. It'd prefer just be extra safe for the reasonable low cost of an equality check.

Note: an `isclose` call is probably most appropriate here since we're doing a variety of str conversions.